### PR TITLE
Add XLOOKUP without support for non standard match_mode and search_mode - and returning values/arrays instead of ranges

### PR DESCRIPTION
**Pull request by selimyoussry**
### Context

Per [this discussion](https://github.com/handsontable/hyperformula/discussions/1064) it seems like XLOOKUP is useful to a bunch of people. It is to for me at the very least :) and so I am happy to contribute to its implementation!

### How did you test your changes?

I have tested the XLOOKUP implementation against the examples from [Microsoft's official doc](https://support.microsoft.com/en-us/office/xlookup-function-b7fd680e-6d10-43e6-84f9-88eae8bf5929)

- [x] Example 1: should find value in simple column range
- [x] Example 2:  should find row range in table
- [x] Example 2 transposed: should find column range in table (official example 2, transposed). This makes sure both column and row search work.
- [x] Example 3: should find use if_not_found argument if not found
- [ ] Example 4: current limitation of this implementation, the approximate match and non standard search are not supported. I added these limitations to the documentation. Hence, this example is not tested against.
- [x] Example 5: nested xlookup function to perform both a vertical and horizontal match (official example 5)
- [ ] Example 6: two nested xlookup + sum (official example 6). I wrote the code but then realized that (based on my shallow understanding of Hyperformula, please correct me if I am wrong), Hyperformula does not support functions returning ranges, only functions returning values.

### Types of changes

- [ ] Breaking change (a fix or a feature because of which an existing functionality doesn't work as expected anymore)
- [x] New feature or improvement (a non-breaking change that adds functionality)
- [ ] Bug fix (a non-breaking change that fixes an issue)
- [ ] Additional language file, or a change to an existing language file (translations)
- [ ] Change to the documentation

### Related issues:
- Related to this discussion https://github.com/handsontable/hyperformula/discussions/1064

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to HyperFormula](https://hyperformula.handsontable.com/guide/contributing.html) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://goo.gl/forms/yuutGuN0RjsikVpM2).
- [ ] My change is compliant with the [OpenDocument](https://docs.oasis-open.org/office/OpenDocument/v1.3/os/part4-formula/OpenDocument-v1.3-os-part4-formula.html) standard. **This does not apply as XLOOKUP is not part of this standard**
- [x] My change is compatible with Microsoft Excel.
- [x] My change is compatible with Google Sheets. **Should be if the XLOOKUP implementation in Sheets is the same as Excel**
- [ ] I described my changes in the [CHANGELOG.md](https://github.com/handsontable/hyperformula/blob/master/CHANGELOG.md) file. **That feels like it's up to you guys to see if that's releasable :)**
- [x] My changes require a documentation update.
- [ ] My changes require a migration guide.

### CODE REVIEW

**Comment by AMBudnik** [comment]
Thank you for signing the CLA and sharing the pull request.

File: docs/guide/built-in-functions.md [hunk 1]
```diff
@@ -227,6 +227,7 @@ Total number of functions: **{{ $page.functionsCount }}**
...
 | ROWS        | Returns the number of rows in the given reference.                                                                                       | ROWS(Array)                                                            |
 | VLOOKUP     | Searches vertically with reference to adjacent cells to the right.                                                                       | VLOOKUP(Search_Criterion, Array, Index, Sort_Order)                    |
+| XLOOKUP     | The XLOOKUP function searches a range or an array, and then returns the item corresponding to the first match it finds. If no match exists, then XLOOKUP can return the closest (approximate) match. Current limitations: only default match_mode and search_mode are supported, a range of value is returned, not a range, so having XLOOKUP(...):XLOOKUP(...) will not work.                                                                      | XLOOKUP(lookup_value, lookup_array, return_array, [if_not_found], [match_mode], [search_mode])                    |
```
Comment (sequba) [inline docs/guide/built-in-functions.md:230]: I can understand why you put it here, but the current convention in HyperFormula is to describe the limitations of the built-in functions in pages https://hyperformula.handsontable.com/guide/list-of-differences.html (difference in results from other spreadsheets) and https://hyperformula.handsontable.com/guide/known-limitations.html#nuances-of-the-implemented-functions (general limitations).

File: src/i18n/languages/trTR.ts [hunk 2]
```diff
@@ -232,6 +232,7 @@ const dictionary: RawTranslationPackage = {
     WEEKNUM: 'HAFTASAY',
     WORKDAY: 'İŞGÜNÜ',
     'WORKDAY.INTL': 'İŞGÜNÜ.ULUSL',
+    XLOOKUP: 'ÇAPRAZARA',
```
Comment (sequba) [inline src/i18n/languages/trTR.ts:235]: Thank you for providing the translations. We really appreciate it.

File: src/interpreter/plugin/LookupPlugin.ts [hunk 3]
```diff
@@ -94,6 +116,65 @@ export class LookupPlugin extends FunctionPlugin implements FunctionPluginTypech
     })
   }
 
+  /**
+   * Corresponds to XLOOKUP(lookup_value, lookup_array, return_array, [if_not_found], [match_mode], [search_mode])
+   * 
+   * @param ast
+   * @param state
+   */
+  public xlookup(ast: ProcedureAst, state: InterpreterState): InterpreterValue {
```
Comment (sequba) [inline src/interpreter/plugin/LookupPlugin.ts:125]: Very solid implementation 💪

**Review by sequba — COMMENTED** [review]
Very solid implementation. Thanks a lot, @selimyoussry. We appreciate that you invested your time to contribute to the HyperFormula.

It requires polishing and probably some of the limitations you described can be addressed to make the function more robust, but it's a good starting point and certainly the right direction to go about implementing the XLOOKUP function.

**Comment by sequba** [comment]
Merging this PR into a local branch to fine-tune the implementation. Thank you, @selimyoussry, for the contribution.

**Comment by AMBudnik** [comment]
Thank you again for your input, @selimyoussry, and for spending your time improving HyperFormula. I'm glad to see the new function added in HF v3. Thank you!

**Pull request merged**